### PR TITLE
cleanup the callers of the resamplers for mic to net and net to speaker, remove loopback resampler

### PR DESCRIPTION
**Pull request by ZappoMan**
We don't want to ever allow the OS/Qt to "claim" to support our sample rate, since it doesn't actually handle it. We've had this dead code that is confusing to read, so this PR just cleans that up so the code reads like what it actually does. This does not change functionality in any way.

### CODE REVIEW

**Comment by ZappoMan** [comment]
Test plan:
- smoke test
- make sure audio works
- test local loopback
- test server loopback
- test on Mac and Windows

This is a client only change, and should work with any domain.

File: libraries/audio-client/src/AudioClient.cpp [hunk 1]
```diff
@@ -315,54 +314,35 @@ bool adjustedFormatForAudioDevice(const QAudioDeviceInfo& audioDevice,
...
+    const int HALF_FORTY_FOUR = FORTY_FOUR / 2;
+
+    if (audioDevice.supportedSampleRates().contains(AudioConstants::SAMPLE_RATE * 2)) {
+        // use 48, which is a sample downsample, upsample
+        adjustedAudioFormat.setSampleRate(AudioConstants::SAMPLE_RATE * 2);
+    } else if (audioDevice.supportedSampleRates().contains(HALF_FORTY_FOUR)) {
+        // use 22050, resample but closer to 24
+        adjustedAudioFormat.setSampleRate(HALF_FORTY_FOUR);
+    } else if (audioDevice.supportedSampleRates().contains(FORTY_FOUR)) {
+        // use 48000, resample
```
Comment (kencooke) [inline libraries/audio-client/src/AudioClient.cpp]: comment looks wrong, should be 44100

File: libraries/audio-client/src/AudioClient.cpp [hunk 2]
```diff
@@ -315,54 +314,35 @@ bool adjustedFormatForAudioDevice(const QAudioDeviceInfo& audioDevice,
...
-            // use 22050, resample but closer to 24
-            adjustedAudioFormat.setSampleRate(HALF_FORTY_FOUR);
-        } else if (audioDevice.supportedSampleRates().contains(FORTY_FOUR)) {
-            // use 48000, resample
-            adjustedAudioFormat.setSampleRate(FORTY_FOUR);
-        }
+    const int HALF_FORTY_FOUR = FORTY_FOUR / 2;
+
+    if (audioDevice.supportedSampleRates().contains(AudioConstants::SAMPLE_RATE * 2)) {
+        // use 48, which is a sample downsample, upsample
```
Comment (kencooke) [inline libraries/audio-client/src/AudioClient.cpp]: should be "simple" instead of sample

**Comment by hifi-gustavo** [comment]
Build console output for commit: c0be9aec6aedd683e12d7d61dfbf9ff210526d70 available at: https://gist.github.com/cd5ae3431d0f983665e406711105f226<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8125.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8125.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 6111e4952afdfced12aca6ac0eb1c2d1906793d8 available at: https://gist.github.com/cd5ae3431d0f983665e406711105f226<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8125.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8125.dmg

**Comment by OmegaHeron** [comment]
Testing Windows 10
Local audio loopback - pass
Server audio loopback - pass
Regular audio via server -pass
Sounds in various scenes like "home", playa - pass
No Mac test possible - don't have one.

**Comment by OmegaHeron** [comment]
Testing Windows 10
Local audio loopback - pass
Server audio loopback - pass
Regular audio via server -pass
Sounds in various scenes like "home", playa - pass
No Mac test possible - don't have one.

**Comment by kencooke** [comment]
Tested on Windows 8.1 - PASS
No Mac available.

Local, network, local echo, and server echo all work as expected.
The log as expected:

[06/23 11:35:28] [DEBUG] The default audio input device is "Analog (1+2) (RME Babyface)"
[06/23 11:35:28] [DEBUG] The audio input device  "Analog (1+2) (RME Babyface)" is available.
[06/23 11:35:28] [DEBUG] The desired format for audio I/O is QAudioFormat(24000Hz, 16bit, channelCount=1, sampleType=SignedInt, byteOrder=LittleEndian, codec="audio/pcm")
[06/23 11:35:28] [DEBUG] The format to be used for audio input is QAudioFormat(48000Hz, 16bit, channelCount=1, sampleType=SignedInt, byteOrder=LittleEndian, codec="audio/pcm")
[06/23 11:35:28] [DEBUG] Attemping to create a resampler for input format to network format.

[06/23 11:35:28] [DEBUG] The default audio output device is "Speakers (RME Babyface)"
[06/23 11:35:28] [DEBUG] The audio output device  "Speakers (RME Babyface)" is available.
[06/23 11:35:28] [DEBUG] The desired format for audio I/O is QAudioFormat(24000Hz, 16bit, channelCount=2, sampleType=SignedInt, byteOrder=LittleEndian, codec="audio/pcm")
[06/23 11:35:28] [DEBUG] The format to be used for audio output is QAudioFormat(48000Hz, 16bit, channelCount=2, sampleType=SignedInt, byteOrder=LittleEndian, codec="audio/pcm")
[06/23 11:35:28] [DEBUG] Attemping to create a resampler for network format to output format.

**Comment by kencooke** [comment]
Verified to work on Mac.

**Pull request merged**